docs/docs/tutorial/creating-a-source-plugin/part-8/index.mdx

```diff
@@ -0,0 +1,196 @@
+
+    ```json:title=package.json
+    {
+      "version": "1.0.0",
+    }
+    ```
+- Include a `keywords` field in your plugin's `package.json`, containing `gatsby` and `gatsby-plugin`. This way the plugin can be found through the [plugin library](/plugins/).
+
+    ```json:title=package.json
+    {
+      "keywords": [
+        "gatsby",
+        "gatsby-plugin"
+      ],
+    }
+    ```
+- Include a `peerDependencies` field in your plugin's `package.json`, containing the `gatsby` version range that your plugin is compatible with.
+
+    For example, if your plugin supports Gatsby 5 only (e.g. it uses an API only available in Gatsby 5), use:
+
+    ```json:title=package.json
+    {
+      "peerDependencies": {
+        "gatsby": "^5.0.0"
+      }
+    }
+    ```
+
+    If now Gatsby comes out with a new major version and your plugin didn't use any APIs that needed changes, you could mark your plugin compatible with Gatsby 5 and Gatsby 6 like so:
+
+    ```json:title=package.json
+    {
+      "peerDependencies": {
+        "gatsby": "^5.0.0 || ^6.0.0"
+      }
+    }
+    ```
+- Add a `build` script to your plugin's `package.json`. The `build` script should compile your source code to CommonJS (CJS) into a `dist` folder. If you're authoring the code in TypeScript, also consider generating type definitions. The details of this depend on your [tooling](#tooling).
+
+    This tutorial uses `tsc` so the relevant pieces look something like this:
+
+    ```json:title=package.json
+    {
+      "scripts": {
+        "build": "tsc",
+      }
+    }
+    ```
+
+    ```json:title=tsconfig.json
+    {
+      "include": ["./src/**/*"],
+      "compilerOptions": {
+        "target": "ES6",
+        "lib": ["es2016", "dom"],
+        "esModuleInterop": true,
+        "moduleResolution": "node",
+        "module": "CommonJS",
+        "outDir": "./dist",
+        "declaration": true
+      }
+    }
+    ```
+
+    Since your compiled information will be in the `dist` folder, you need to also add a `main` and `files` key to the `package.json`.
+
+    ```json:title=package.json
+    {
+      "main": "./dist/gatsby-node.js",
+      "files": [
+        "./dist/*",
+        "gatsby-node.js"
+      ],
+    }
+    ```
+
+    If you've generated TypeScript types, consider [adding a `types` key](https://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html).
+
+- Add a `prepare` script to your plugin's `package.json` that runs before `npm publish`. If your `build` script isn't automatically clearing the `dist` folder before doing a new build, add an additional `clean` script. This is to ensure that inside the `dist` folder no old artifacts are being published (e.g. you're renaming a file and without the `clean` the old file would still be published through `dist`). You could e.g. use [del-cli](https://www.npmjs.com/package/del-cli) to achieve this. It would look something like this:
```

Comment:
Possible edit for brevity:
```suggestion
- Add a `prepare` script to your plugin's `package.json` that runs before `npm publish`. If your `build` script isn't automatically clearing the `dist` folder before doing a new build, add an additional `clean` script. This is to ensure that inside the `dist` folder no old artifacts are being published (e.g. you're renaming a file and without the `clean` the old file would still be published through `dist`). You could use [del-cli](https://www.npmjs.com/package/del-cli) to achieve this. It would look something like this:
```